query/logicalplan/optimize.go

```diff
@@ -17,19 +21,31 @@ var DefaultOptimizers = []Optimizer{
 // the query.
 type PhysicalProjectionPushDown struct{}
 
-func (p *PhysicalProjectionPushDown) Optimize(plan *LogicalPlan) *LogicalPlan {
-	p.optimize(plan, nil)
+func (p *PhysicalProjectionPushDown) Optimize(schema *dynparquet.Schema, plan *LogicalPlan) *LogicalPlan {
+	p.optimize(schema, plan, nil)
 	return plan
 }
 
-func (p *PhysicalProjectionPushDown) optimize(plan *LogicalPlan, columnsUsed []ColumnMatcher) {
+func (p *PhysicalProjectionPushDown) optimize(schema *dynparquet.Schema, plan *LogicalPlan, columnsUsed []ColumnMatcher) {
 	switch {
 	case plan.SchemaScan != nil:
 		plan.SchemaScan.Projection = columnsUsed
 	case plan.TableScan != nil:
 		plan.TableScan.Projection = columnsUsed
 	case plan.Filter != nil:
-		columnsUsed = append(columnsUsed, plan.Filter.Expr.ColumnsUsed()...)
+		columns := plan.Filter.Expr.ColumnsUsed()
+		for _, column := range columns {
+			if dc, ok := schema.ColumnByName(column.Name()); ok {
+				if dc.Dynamic {
+					columnsUsed = append(columnsUsed, DynamicColumnMatcher{ColumnName: column.Name()})
```

Comment:
Is there a specific reason why we're doing this? I don't understand what a filter of a dynamic column could look like, as a `DynamicColumnMatcher` is intended to match any of the dynamically created columns.